ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -646,24 +679,37 @@ public String getResourceNameForPartition(AmbryPartition partition) {
 
     /**
      * {@inheritDoc}
-     * If dcName is null, then get replicas by given state from all datacenters.
+     * If dcOrClusterName is null, then get replicas by given state from all datacenters.
      * If no routing table snapshot is found for dc name, or no resource name found for given partition, return empty list.
      */
     @Override
     public List<AmbryReplica> getReplicaIdsByState(AmbryPartition partition, ReplicaState state, String dcName) {
       List<AmbryReplica> replicas = new ArrayList<>();
-      for (DcInfo dcInfo : dcToDcInfo.values()) {
-        String dc = dcInfo.dcName;
-        if (dcName == null || dcName.equals(dc)) {
-          String resourceName = partitionToResourceNameByDc.get(dc).get(partition.toPathString());
-          dcToRoutingTableSnapshotRef.get(dcInfo.dcName)
-              .get()
-              .getInstancesForResource(resourceName, partition.toPathString(), state.name())
-              .stream()
-              .map(instanceConfig -> instanceNameToAmbryDataNode.get(instanceConfig.getInstanceName()))
-              .map(dataNode -> ambryDataNodeToAmbryReplicas.get(dataNode).get(partition.toPathString()))
-              .filter(Objects::nonNull)
-              .forEach(replicas::add);
+      if (clusterMapConfig.clusterMapUseAggregatedView) {
+        String resourceName = dcName != null ? partitionToResourceNameByDc.get(dcName).get(partition.toPathString())
```

Comment:
does this mean that all datacenters have the same resourceName for a partition. I suppose they should be the same, we only get on resourceName for each partition in clustermap.

Reply:
Thank you for pointing it out. Actually it looks like it is not guaranteed that partitions would be mapped to same resource in each data center (found some examples in zookeeper). Made changes to go through resource names of all DCs and get replicas accordingly.